### PR TITLE
Apply Refactored Custom Tooltip

### PR DESCRIPTION
**Pull request by joeizang**
Replace the raw react bootstrap components with our own component that encapsulates all that complexity as prescribed by @ioedeveloper

### CODE REVIEW

File: apps/remix-ide-e2e/src/tests/proxy.test.ts [hunk 1]
```diff
@@ -70,12 +70,13 @@ module.exports = {
     browser
       .openFile('myTokenV1.sol')
       .clickLaunchIcon('solidity')
-      .pause(2000)
+      .pause(5000)
```
Comment (bunsenstraat) [inline apps/remix-ide-e2e/src/tests/proxy.test.ts:73]: Don't add pauses, if you can use assertions to see if the element is present.same for the other pause.

File: libs/remix-ui/debugger-ui/src/lib/button-navigator/button-navigator.tsx [hunk 2]
```diff
@@ -55,113 +55,125 @@ export const ButtonNavigation = ({ stepOverBack, stepIntoBack, stepIntoForward,
   const disableStepBtnStyle = 'stepButtonDisabled'
   const jumpBtnStyle = 'd-flex align-items-center justify-content-center btn btn-primary btn-sm jumpButton h-75 m-0 p-1'
   const disableJumpBtnStyle = 'jumpButtonDisabled'
+  const stepMarkupStructure = {
```
Comment (ioedeveloper) [inline libs/remix-ui/debugger-ui/src/lib/button-navigator/button-navigator.tsx:58]: Can you explain these changes here?

File: yarn.lock [hunk 3]
```diff
@@ -4276,6 +4276,11 @@
   resolved "https://registry.yarnpkg.com/@types/aria-query/-/aria-query-4.2.2.tgz#ed4e0ad92306a704f9fb132a0cfcf77486dbe2bc"
   integrity sha512-HnYpAE1Y6kRyKM/XkEuiRQhTHvkzMBurTHnpFLYLBGPIylZNPs9jJcuOOYWxPLJCSEtmZT0Y8rHDokKN7rRTig==
 
+"@types/async@^3.2.15":
```
Comment (ioedeveloper) [inline yarn.lock:4279]: Do you need changes to `yarn.lock` file?

File: workspace.json [hunk 4]
```diff
@@ -1540,7 +1540,7 @@
               "apps/vyper/src/styles.css"
             ],
             "scripts": [],
-            "webpackConfig": "@nrwl/react/plugins/webpack"
+            "webpackConfig": "apps/vyper/webpack.config.js"
```
Comment (Aniket-Engg) [inline workspace.json:1543]: Is this change necessary for this PR?

File: libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx [hunk 5]
```diff
@@ -549,25 +557,52 @@ export function ContractGUI (props: ContractGUIProps) {
...
                 !useLastProxy ?
                 <div className="mb-2">
-                  <label className='mt-2 text-left d-block'>Proxy Address: </label>
-                  <input style={{ height: 32 }} className="form-control udapp_input" data-id="ERC1967AddressInput" placeholder='proxy address' title='Enter previously deployed proxy address on the selected network' onChange={handleSetProxyAddress} onBlur={() => validateProxyAddress(proxyAddress) } />
-                  { proxyAddressError && <span className='text-lowercase' data-id="errorMsgProxyAddress" style={{ fontSize: '.8em' }}>{ proxyAddressError }</span> }
-                </div> :
-                <span className='text-capitalize' data-id="lastDeployedERC1967Address" style={{ fontSize: '.8em' }}>{ proxyAddress || proxyAddressError }</span>
-              }
+                  <label className="mt-2 text-left d-block">
+                    Proxy Address:{" "}
```
Comment (ioedeveloper) [inline libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx:580]: I think this is an error in this line `{" "}`

File: libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx [hunk 6]
```diff
@@ -446,14 +435,21 @@ export function ContractGUI (props: ContractGUIProps) {
...
               />
-              <label
-                htmlFor="deployWithProxy"
-                data-id="contractGUIDeployWithProxyLabel"
-                className="m-0 form-check-label custom-control-label udapp_checkboxAlign"
-                title="An ERC1967 proxy contract will be deployed along with the selected implementation contract."
-              >
-                Deploy with Proxy
-              </label>
+              {/* <CustomTooltip
```
Comment (ioedeveloper) [inline libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx:438]: Can you please remove commented lines?

File: libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx [hunk 7]
```diff
@@ -489,15 +485,21 @@ export function ContractGUI (props: ContractGUIProps) {
...
-                      <input
-                        ref={(el) => {
-                          initializeFields.current[index] = el;
-                        }}
-                        style={{ height: 32 }}
-                        className="form-control udapp_input"
-                        placeholder={inp.type}
-                        title={inp.name}
-                      />
+                      {/* <CustomTooltip
```
Comment (ioedeveloper) [inline libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx:488]: Here also.

**Review by ioedeveloper — CHANGES_REQUESTED** [review]

File: yarn.lock (hunk 3 above)
Comment (joeizang) [reply yarn.lock:4279]: These are changes that came from the fix I got from @bunsenstraat.

File: libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx (hunk 6 above)
Comment (joeizang) [reply libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx:438]: Done

File: libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx (hunk 5 above)
Comment (joeizang) [reply libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx:580]: Yes This was a mistake with prettier. Fixed

File: libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (joeizang) [reply libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx:438]: Done

File: libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx (hunk 7 above)
Comment (joeizang) [reply libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx:488]: Done

File: workspace.json (hunk 4 above)
Comment (joeizang) [reply workspace.json:1543]: Yes, vyper plugin was failing e2e. @bunsenstraat helped fix the issue which had to do with webpack.

File: apps/remix-ide-e2e/src/tests/proxy.test.ts (hunk 1 above)
Comment (joeizang) [reply apps/remix-ide-e2e/src/tests/proxy.test.ts:73]: Ok. Thanks

**Review by ioedeveloper — APPROVED** [review]
